src/sagemaker_sklearn_extension/feature_extraction/sequences.py

```diff
@@ -0,0 +1,193 @@
+            # Convert inf (if any) to np.nan
+            numeric_sequence = [value if (not np.isinf(value)) else np.nan for value in numeric_sequence]
+            numeric_sequences.append(numeric_sequence)
+        # Pad shorter sequences with np.nan to reach the length of the longest sequence
+        max_length = np.max([len(numeric_sequence) for numeric_sequence in numeric_sequences])
+        padded_sequences = []
+        for sequence in numeric_sequences:
+            if len(sequence) < max_length:
+                length_difference = max_length - len(sequence)
+                padding_nans = np.array([np.nan] * length_difference)
+                sequence = np.hstack([sequence, padding_nans])
+            padded_sequences.append(sequence)
+        X = np.array(padded_sequences)
+        return X
+
+    def _more_tags(self):
+        return {"X_types": ["string"], "allow_nan": True}
+
+
+class TSFreshFeatureExtractor(BaseEstimator, TransformerMixin):
+    """Extract features computed by tsfresh from each row of an array and append them to the input
+    array (if augment = True) or return the extracted features alone (if augment = False).
+
+    Examples of these features are the mean, median, kurtosis, and autocorrelation of each sequence. The full list
+    of extracted features can be found at https://tsfresh.readthedocs.io/en/latest/text/list_of_features.html.
+
+    Parameters
+    ----------
+    augment : boolean (default=True):
+        Whether to append the tsfresh features to the original raw data (if True),
+        or output only the extracted tsfresh features (if False)
+
+    interpolation_method : {'linear', 'fill', 'zeroes', None} (default='zeroes')
+        'linear': linear interpolation
+        'fill': forward fill to complete the sequences; then, backfill in case of NaNs at the start of the sequence.
+        'zeroes': pad with zeroes
+         None: no interpolation
+
+    Attributes
+    ----------
+    self.robust_standard_scaler_ : ``sagemaker_sklearn_extension.preprocessing.data.RobustStandardScaler``
+        - `robust_standard_scaler_` is instantiated inside the fit method used for computing the mean and
+        the standard deviation.
+
+    """
+
+    def __init__(self, augment=True, interpolation_method="zeroes"):
+        super().__init__()
+        self.augment = augment
+        self.interpolation_method = interpolation_method
+
+    def fit(self, X, y=None):
+        X = check_array(X, force_all_finite="allow-nan")
+        self._dim = X.shape[1]
+        tsfresh_features, _ = self._extract_tsfresh_features(X)
+        robust_standard_scaler = RobustStandardScaler()
+        robust_standard_scaler.fit(tsfresh_features)
+        self.robust_standard_scaler_ = robust_standard_scaler
+        return self
+
+    def transform(self, X, y=None):
+        X = check_array(X, force_all_finite="allow-nan")
```

Comment:
Is X an np.array?
I find it odd that we are forcing the inputs for feature extraction to have the same dimension. Does this mean that before using tsfresh to extract features, I'll pad all the short sequences with nan, then impute missing values, and only then call tsfresh?
Or will tsfresh know it should ignore the nans at the end of a sequence?

Reply:
Correct, X is a np.array. I've added docstrings for clarity. 

Indeed, we impute missing values through X_df = self._interpolate(X_df) as part of TSFreshFeatureExtractor() before extracting features with tsfresh. We do this as tsfresh is designed not to accept nans in the input.

Reply:
Here, I'm a little concerned. Consider a dataset with 1000 seuqences of length 3, and a single sequence of length 20.
This single long sequence will considerably change the features extracted with tsfresh, causing our algorithm to be somewhat unstable. What's worse is that this transform operation is done in mini-batches for large dataset. We can have the same exact row appearing in different batches and becoming a different vector. 

If its not hurting performance (speed), I think it would be cleaner to apply tsfresh on the original vectors without padding + missing value imputation

Reply:
I've addressed this in the revision. TSFreshFeatureExtractor now works with a list of (possibly variable-length) arrays and extracts features directly from the original vectors. If the user requires the tsfresh features to be stacked to the original sequences (augment = True), we pad and impute the original sequences only before stacking.